internal/attestations/attestations.go

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package attestations
+
+import (
+	"errors"
+
+	"github.com/gittuf/gittuf/internal/gitinterface"
+	"github.com/gittuf/gittuf/internal/rsl"
+	"github.com/gittuf/gittuf/internal/third_party/go-git"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing/filemode"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing/object"
+)
+
+const (
+	Ref                         = "refs/gittuf/attestations"
+	authorizationsTreeEntryName = "authorizations"
+	initialCommitMessage        = "Initial commit"
+	defaultCommitMessage        = "Update attestations"
+)
+
+var ErrAttestationsExist = errors.New("cannot initialize attestations namespace as it exists already")
+
+// InitializeNamespace creates a namespace to store attestations for
+// verification with gittuf. The ref is created with an initial, unsigned commit
+// that is unsigned.
+func InitializeNamespace(repo *git.Repository) error {
+	if ref, err := repo.Reference(plumbing.ReferenceName(Ref), true); err != nil {
+		if !errors.Is(err, plumbing.ErrReferenceNotFound) {
+			return err
+		}
+	} else if !ref.Hash().IsZero() {
+		return ErrAttestationsExist
+	}
+
+	treeHash, err := gitinterface.WriteTree(repo, nil)
+	if err != nil {
+		return err
+	}
+
+	_, err = gitinterface.Commit(repo, treeHash, Ref, initialCommitMessage, false)
+	return err
+}
+
+// Attestations tracks all the attestations in a gittuf repository.
+type Attestations struct {
+	authorizations map[string]plumbing.Hash // refPath/from-to -> blob ID
```

Comment:
It would be also useful to add more documentation (either README.md or godoc) on the Attestation format and how refs, commits, ReferenceEntries, and attestations relate to each other.

Reply:
Updated the comment to be more complete of the specific implementation. I'll add authorizations more generally to the design doc, overriding #68.

Reply:
https://github.com/gittuf/gittuf/pull/195